pkg/cachemanager/cachemanager_test/cachemanager_integration_test.go

```diff
@@ -35,92 +35,137 @@ const (
 
 var cfg *rest.Config
 
+var (
+	configMapGVK = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ConfigMap"}
+	podGVK       = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"}
+
+	configInstancOne      = "config-test-1"
+	configInstancTwo      = "config-test-2"
+	nsedConfigInstanceOne = "default/config-test-1"
+	nsedConfigInstanceTwo = "default/config-test-2"
+
+	podInstanceOne     = "pod-test-1"
+	nsedPodInstanceOne = "default/pod-test-1"
+)
+
 func TestMain(m *testing.M) {
 	testutils.StartControlPlane(m, &cfg, 3)
 }
 
-// TestCacheManager_replay_retries tests that we can retry GVKs that error out in the reply goroutine.
+type failureInjector struct {
```

Comment:
This is really cool!  

It seems like it could be well-homed alongside the SpyReader, since they are made to be used together.

Reply:
moved next to the `SpyReader`.